File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    closet = {}
    for name, kind in clothes:
        if kind in closet.keys():
            closet[kind] += [name]
        else:
            closet[kind] = [name]
            
    for _, value in closet.items():
        answer *= (len(value) + 1)
        
    return (answer - 1)

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string,int> m;
    for(auto c: clothes) {
        string type = c[1];
        m[type]++;
    }
    for(auto [key,val]: m) {
        answer += answer * val;
    }
    return answer-1;
}